slycot/src/math.pyf

```diff
@@ -14,9 +14,9 @@ end subroutine mc01td
 
 subroutine mb02ed(typet,k,n,nrhs,t,ldt,b,ldb,dwork,ldwork,info) ! in MB02ED.f
   character :: typet
-  integer intent(in) :: k
+  integer intent(in),required,check(k>=0)  :: k
```

Comment:
Too many safeguards for the same thing. Through adding a `check` in the `pyf`, you never get to the Fortran routine generating `INFO=-1`.